Add option to configure seLinux and runAs policies

AlbertoPimpo opened the pull request:
EDIT: after discussions, we decided to make seLinux and runAs policies configurable instead of changing the default.

When creating a securityContextContraint for Openshift, the helm chart currently set the selinux strategy to `MustRunAs`, but this might lead to permission errors if you are using selinux in the node and you want to mount an host folder with an HostPath.

I think there is no reason for disallowing selinux context to be changed for the fluentbit deployment. Also, this is the same behavior used by red hat for their cluster logging operator that deploys fluentd.
https://github.com/openshift/cluster-logging-operator/blob/master/internal/auth/securitycontextconstraint.go#L46

AlbertoPimpo commented:
Sorry for all the pushes but I had to fix the verification of the commit :sweat_smile:

AlbertoPimpo commented:
@stevehipwell @naseemkullah @edsiper could you please take a look?

stevehipwell commented:
@AlbertoPimpo I'm not familiar enough with OpenShift to make a call on this change.

FYI you need to rebase and bump your chart version.

AlbertoPimpo commented:
> FYI you need to rebase and bump your chart version

@stevehipwell done :+1:

AlbertoPimpo commented:
If it can help, in the PodSecurityPolicy you indeed allow to change the selinux context https://github.com/fluent/helm-charts/blob/main/charts/fluent-bit/templates/psp.yaml#L28

patrick-stephens commented:
I would not be a fan of changing it from one hardcoded value to another, let's make it configurable with a default.

AlbertoPimpo commented:
> I would not be a fan of changing it from one hardcoded value to another, let's make it configurable with a default.

@patrick-stephens I like the idea, maybe we should do the same for psp? I can take care of that.

AlbertoPimpo commented on charts/fluent-bit/values.yaml:
```diff
@@ -54,6 +59,10 @@ openShift:
+    runAsUser:
+      type: RunAsAny
+    seLinuxContext:
+      type: MustRunAs
```
I left the default to `MustRunAs` since most of the users will not need to change the selinux context.

AlbertoPimpo commented:
@patrick-stephens sorry for taking so long, it's done now. Feel free to merge it.

AlbertoPimpo commented:
@stevehipwell @naseemkullah @edsiper this is ready to be merged.

stevehipwell reviewed (changes requested):
@AlbertoPimpo PSPs have not been supported in Kubernetes since `1.26` so I don't see any value in making changes there?

stevehipwell commented on charts/fluent-bit/values.yaml:
```diff
@@ -45,6 +45,11 @@ rbac:
   create: false
   annotations: {}
+  runAsUser:
+    rule: 'RunAsAny'
```
```suggestion
    rule: RunAsAny
```

stevehipwell commented on charts/fluent-bit/values.yaml:
```diff
@@ -45,6 +45,11 @@ rbac:
+    rule: 'RunAsAny'
+  seLinux:
+    # This policy assumes the nodes are using AppArmor rather than SELinux.
+    rule: 'RunAsAny'
```
```suggestion
    rule: RunAsAny
```

stevehipwell commented:
@AlbertoPimpo could you please rebase, look at the comments I just made and also update the chart changelog annotations? As the PSP resource is still in the chart I can't see an issue with making this change.

AlbertoPimpo commented:
> @AlbertoPimpo could you please rebase, look at the comments I just made and also update the chart changelog annotations? As the PSP resource is still in the chart I can't see an issue with making this change.

Thanks for the review @stevehipwell . I accepted your suggested changes, rebased and squashed everything in a single commit. We should be ready to merge.

stevehipwell commented:
@AlbertoPimpo I can't see why the tests on your branch are failing.

stevehipwell reviewed (approved):
LGTM
